user/yaml/kubestellar-server-ingress.yaml

```diff
@@ -0,0 +1,147 @@
+# Copyright 2023 The KubeStellar Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: kubestellar
+
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: kubestellar-pvc
+  namespace: kubestellar
+spec:
+  accessModes:
+  - ReadWriteOnce
+  resources:
+    requests:
+      storage: 1Gi
+
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: kubestellar-service-account
+  namespace: kubestellar
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: kubestellar-role
+  namespace: kubestellar
+rules:
+- resources: [ "secrets" ]

```

Comment:
The principle of least privilege would say that this rule should focus on the one individual Secret object involved.  https://github.com/kubernetes/api/blob/v0.24.3/rbac/v1/types.go#L62 can focus on one object.

Reply:
@MikeSpreitzer can you helm me more specifically, do you want me to just name `kubestellar` secret and no other. I don't know how to do it.

Reply:
Yes, I am trying to suggest to restrict this role to just the one Secret object that is needed.  I gave a pointer to the part of the datatype declaration where individuals can be named.  What it is saying is that to restrict the PolicyRule to just one object named "kubestellar" you would add `resourceNames: [ kubestellar ]` (for a one-line example) alongside where you already have `resources: [ secrets ]`.

Reply:
@MikeSpreitzer thank you for the explanation, I have also updated the role for ingress/route correspondingly.